### PR TITLE
Title validator

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Description
Add title validator

## Testing
good data unit test
Screen Shot 2020-02-04 at 3 01 05 PM

bad data unit test

[*] Running Integration Validators...
[*] Executing validator WorkflowTitleValidator
Validator WorkflowTitleValidator failed!
Title must not be blank
[*]Workflow_no_title failed validation!

----
[*] Total time elapsed: 10.577ms
[*] Running Integration Validators...
[*] Executing validator WorkflowTitleValidator
Validator WorkflowTitleValidator failed!
Title ends with period when it should not.
[*]Workflow_ends_with_period failed validation!

----
[*] Total time elapsed: 9.332999999999998ms
[*] Running Integration Validators...
[*] Executing validator WorkflowTitleValidator
Validator WorkflowTitleValidator failed!
Title contains a lowercase 'indicator' when it should not.
[*]Workflow_bad_caps failed validation!

----
[*] Total time elapsed: 7.132000000000001ms

Testing updated puligin title validator
Screen Shot 2020-02-04 at 3 05 46 PM

### CODE REVIEW

File: icon_validator/rules/workflow_validators/workflow_change_log_validator.py [hunk 1]
```diff
@@ -47,6 +59,9 @@ def validate_order(versions_history):
 
     @staticmethod
     def validate_version_history_updated(versions_history, spec):
+        """
+
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_change_log_validator.py]: Missing docstring

File: icon_validator/rules/workflow_validators/workflow_extension_validator.py [hunk 2]
```diff
@@ -6,11 +6,17 @@ class WorkflowExtensionValidator(KomandPluginValidator):
...
+        """
+        Checks that the extension key in .yaml has a value of workflow
+        """
         if not extension == "workflow":
             raise ValidationException("Extension key must have a value of workflow.")
 
     @staticmethod
     def validate_extension_exists(spec):
+        """
+        hecks that the extension key in .yaml exists.
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_extension_validator.py]:
```suggestion
        Checks that the extension key in .yaml exists.
```

File: icon_validator/rules/workflow_validators/workflow_help_validator.py [hunk 3]
```diff
@@ -6,16 +6,26 @@ class WorkflowHelpValidator(KomandPluginValidator):
 
     @staticmethod
     def validate_help_exists(spec):
+        """
+        Checks that 'help' is not in .yaml
+        In old code help was included in the yaml. It should no longer be their
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_help_validator.py]:
```suggestion
        In old code help was included in the yaml. It should no longer be there
```

File: icon_validator/rules/workflow_validators/workflow_screenshot_validator.py [hunk 4]
```diff
@@ -13,6 +13,14 @@ def __init__(self):
 
     @staticmethod
     def validate_title(title):
+        """
+        Checks that title is not blank.
+        Checks that title dose not end with a period.
+        Checks that title dose not start with a lower case letter.
+        Checks that title dose not start with a space.
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_screenshot_validator.py]: Typo, "does"

File: icon_validator/rules/workflow_validators/workflow_title_validator.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class WorkflowTitleValidator(KomandPluginValidator):
+
+    def validate(self, spec):
+        """
+        Checks that title is not blank.
+        Checks that title dose not end with a period.
+        Checks that title dose not start with a lower case letter.
+        Checks that title dose not start with a space.
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_title_validator.py]: Typo, "does"

File: icon_validator/rules/workflow_validators/workflow_title_validator.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        if title.endswith("."):
+            raise ValidationException("Title ends with period when it should not.")
+        if title[0].islower():
+            # This plugin title is OK: minFraud
+            # This plugin title is OK: ifconfig.co
+            raise ValidationException("Title should not start with a lower case letter.")
+        if title[0].isspace():
+            raise ValidationException("Title should not start with a whitespace character.")
+        if len(title.split()) > 7:
+            raise ValidationException(f"Title is too long, 6 words or less: contains {str(len(title.split()))}")
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_title_validator.py]:
```suggestion
            raise ValidationException(f"Title is too long, 6 words or less: contains {title.count(" ")}")
```

Simpler and no need to wrap the count in a string.

File: icon_validator/rules/workflow_validators/workflow_vendor_validator.py [hunk 7]
```diff
@@ -6,6 +6,12 @@ class WorkflowVendorValidator(KomandPluginValidator):
 
     @staticmethod
     def validate_vendor(vendor):
+        """
+        Check that vendor is not komand.
+        Check that vendor does not end with a period.
+        Check that vendor dose not start with a capital letter.
+        Check that vendor dose not contain spaces.
```
Comment (lmilby-r7) [inline icon_validator/rules/workflow_validators/workflow_vendor_validator.py]: typo "dose"

**Pull request merged**